docs/guides/styles-and-resources/resources.md

```diff
@@ -284,10 +284,54 @@ public MainWindow()
     // found2 = true | result2 = "Hello World" 
     var found2 = this.TryFindResource("TheKey", this.ActualThemeVariant, out var result2);
 
-    // Dound the resource to a TextBlock from code behind
+    // bind the resource to a TextBlock from code behind
     myTextBlock.Bind(TextBlock.TextProperty, Resources.GetResourceObservable("TheKey"));
 
     // this will update myTextBlock.Text via the bound observable
     this.Resources["TheKey"] = "Hello from code behind"; 
 }
 ```
+
+### Global Resource Lookup
+
+In other XAML UI frameworks such as WPF and WinUI, resource lookup could also be done using an indexer and code such as `Application.Current.Resources["TheKey"]` (in addition to `FindResource()` or `TryFindResource()`). Resource lookup using the indexer at the application scope would search globally and recursively in all resource dictionaries. You could almost always find a resource using the indexer. _Avalonia UI_ does not support this and will only search the current resource dictionary when using an indexer. Additionally, _Avalonia UI_ may store resources in both resource dictionaries or style resource dictionaries so it's common to be required to search both. For global, recursive resource lookup at the application scope similar to other XAML UI frameworks code such as below may be used.
```

Comment:
But this method is not necessary. Existing TryFindResource should already cover this case. Or am I missing something?

Reply:
TryFindResource() won't search BOTH the resources and style resources (these never should have been separated IMO). It is one or the other depending on where you call it.

Additionally, the indexer functionality is completely different from WPF/WinUI which this section explains as well. The new method reproduces exactly what the old WPF indexer resource lookup logic does. Explained further in #6021 and https://github.com/AvaloniaUI/Avalonia/discussions/6019.

Reply:
> TryFindResource() won't search BOTH the resources and style resources

It will. https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/Application.cs#L215-L216
Controls/Window should also have the same behavior.
Keep in mind, that Fluent/SimpleTheme are styles, that contain resources and it can be reached with TryFindResource 

> Additionally, the indexer functionality is completely different

Yes, that's another problem, that should be mentioned. But TryFindResource specifically works.

Reply:
> Keep in mind, that Fluent/SimpleTheme are styles, that contain resources and it can be reached with TryFindResource

I'll test again but I know for a fact I had to add those lines of code iterating through Style resources to get FluentAvalonia resource lookup to work. So unless something has changed in the last few months I'm not sure I agree that it works.

Reply:
@maxkatz6 Note I'm not the only one that has found issues in this area: https://github.com/amwx/FluentAvalonia/discussions/378#discussioncomment-6030956

Reply:
@maxkatz6 I confirmed again that Styles does need to be searched to pick up resources such as `OverlayCornerRadius` with FluentAvalonia. This indicates that the code snippet I provided is correct.

If you would like I can move all of this to the WPF section: get-started/wpf/styling

Reply:
> these never should have been separated IMO

I can agree that it shouldn't be allowed to add resources to the Style. I.e. remove Style.Resources collection completely. But it's couple of years too late for that.